crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -156,12 +171,36 @@ impl ManifestListWriter {
         match self.format_version {
             FormatVersion::V1 => {
                 for manifest_entry in manifest_entries {
-                    let manifest_entry: ManifestListEntryV1 = manifest_entry.into();
+                    let manifest_entry: ManifestListEntryV1 = manifest_entry.try_into()?;
                     self.avro_writer.append_ser(manifest_entry)?;
                 }
             }
             FormatVersion::V2 => {
-                for manifest_entry in manifest_entries {
+                for mut manifest_entry in manifest_entries {
+                    if manifest_entry.sequence_number == UNASSIGNED_SEQUENCE_NUMBER {
```

Comment:
It seems weird to me to check this in a writer. I think this is supposed to do in transaction api.

Reply:
Yes, we also can do this check outside the writer.🤔 Looks reasonable.

Reply:
Sorry, let's resolve this, I confuse this with manifest